baseplate/lib/experiments/__init__.py

```diff
@@ -29,6 +30,42 @@ class EventType(Enum):
     BUCKET = "choose"
 
 
+class ExperimentsGlobalCache:
+    """Experiment global cache store the parsed experiment.
+
+    Every time gets the global cache, it reads the data from FileWatcher,
+    if file has been updated, it assign its cache to new empty dictionary.
+    Global_cache will be passed to each request's experiment object,
+    parsed experiment will be lazily added when each request parse its experiment
+    if it does not exist in the global_cache.
+    """
+
+    def __init__(self, filewatcher: FileWatcherWithUpdatedFlag):
+        self._filewatcher = filewatcher
+        self._global_cache: Dict[str, Optional[Experiment]] = {}
+
+    def _get_config(self) -> Optional[Dict[str, Dict[str, str]]]:
+        try:
+            config_data: Dict[str, Dict[str, str]]
+            config_data, updated = self._filewatcher.get_data()
+
+            if updated and self._global_cache:
+                self._global_cache = {}
+            return config_data
+        except WatchedFileNotAvailableError as exc:
+            logger.warning("Experiment config unavailable: %s", str(exc))
+            return None
```

Comment:
Would returning `{}` work? It looks like at least some cases down below (like `get_all_experiment_names()`) wouldn't need extra checks anymore if we do that.